docs/deep-learning/lora_training.md

```diff
@@ -54,103 +53,98 @@ class SimpleMLP(nn.Module):
         out = self.fc2(out)
         return out
 
+# Create an initial model
+model = SimpleMLP()
+
+# Apply LoRA configuration
 lora_config = LoraConfig(
-    r=1, lora_alpha=1, lora_dropout=0.01, target_modules=["fc1", "fc2"], bias="none"
+    r=1,
+    lora_alpha=1,
+    lora_dropout=0.01,
+    target_modules=["fc1", "fc2"],
+    bias="none"
 )
 
-model = SimpleMLP()
-# The initial training loop of the model should be
-# added at this point on an initial data-set
+peft_model = get_peft_model(model, lora_config)
 
-# A second data-set, task2 is generated
+# Generate a second data-set for demonstration purposes
 X_task2, y_task2 = make_circles(n_samples=32, noise=0.2, factor=0.5)
 train_loader_task2 = DataLoader(
     TensorDataset(torch.Tensor(X_task2), torch.LongTensor(y_task2)),
     batch_size=32,
     shuffle=True
 )
-
-# Apply LoRA to the model
-peft_model = get_peft_model(model, lora_config)
 ```
 
 ### 2. Convert the LORA model to use custom Concrete ML layers
 
-Concrete ML requires converting the `peft` model to add
-FHE compatible layers. In this step, you can configure several fine-tuning
-parameters:
+Next, we need to integrate the LoRA-adapted `peft_model` into the Concrete ML hybrid FHE training framework. This is done using the `LoraTrainer` class, which handles the logic of encrypting outsourced computations, running the forward and backward passes, and updating the LoRA adapter weights.
 
-- The number of gradient accumulation steps: LORA commonly accumulate gradients over several gradient descent steps before updating weights.
-- The optimizer parameters
-- The loss function
+You can configure:
+
+- The loss function.
+- The optimizer and its parameters.
+- Gradient accumulation steps (if needed).
 
 <!--pytest-codeblocks:cont-->
 
 ```python
-lora_training = LoraTraining(peft_model)
-
-
-# Update training parameters, including loss function
-lora_training.update_training_parameters(
-    optimizer=optim.Adam(filter(lambda p: p.requires_grad, peft_model.parameters()), lr=0.01),
-    loss_fn=nn.CrossEntropyLoss(),
-    training_args={"gradient_accumulation_steps": 1},
+# Define a simple loss function
+def simple_loss(outputs, targets):
+    return F.cross_entropy(outputs, targets)
+
+# Create an Adam optimizer
+optimizer = optim.Adam(peft_model.parameters(), lr=1e-3)
+
+# Initialize trainer with the loss and optimizer
+lora_trainer = LoraTrainer(
+    peft_model,
+    optimizer=optimizer,
+    loss_fn=simple_loss,
 )
-
 ```
 
 ### 3. Compile a hybrid FHE model for the LORA adapted PyTorch model
 
-Compile the hybrid FHE model to convert the selected outsourced layers to use FHE, while the rest will run on the client side. Note that the exchange of encrypted activations and gradients may require significant bandwidth.
+Before training in FHE, we need to compile the model. Compilation calibrates and converts the outsourced linear layers to their FHE equivalents. The compile method uses representative data for this step.
```

Comment:
I suggest:

Before training in FHE, the model must first be compiled. This process calibrates and converts the outsourced linear layers into their FHE equivalents. The compilation step needs representative data to ensure accurate calibration.